aten/src/ATen/native/cuda/FusedSgdKernel.cu

```diff
@@ -94,12 +94,12 @@ struct FusedSgdMathFunctor {
     const auto use_faster_load_store{false};
 #endif
     if (use_faster_load_store) {
-      for (auto i_start = threadIdx.x;
+      for (index_t i_start = threadIdx.x;
            i_start * kILP < n && i_start * kILP < chunk_size;
            i_start += blockDim.x) {
 #pragma unroll
         for (auto i = 0; i < depth; i++) {
-          load_store(r_args[i], args[i], 0, i_start);
+          load_store(r_args[i], args[i], static_cast<index_t>(0), i_start);
```

Comment:
Also for my learning: how does this cast help? Is it better on CUDA for every argument in load_store to have a specified type so it could infer bit widths?